src/main/java/com/descope/sdk/mgmt/impl/AuditServiceImpl.java

```diff
@@ -123,4 +133,21 @@ static class ActualAuditRecord {
   static class ActualAuditSearchResponse {
     List<ActualAuditRecord> audits;
   }
+
+  public void createEvent(AuditCreateRequest request) throws DescopeException {
+    if (request == null) {
+      throw ServerCommonException.invalidArgument("request");
+    }
+    if (StringUtils.isBlank(request.getAction())) {
+      throw ServerCommonException.invalidArgument("request.action");
+    }
+    if (StringUtils.isBlank(request.getActorId())) {
+      throw ServerCommonException.invalidArgument("request.actorId");
+    }
+    if (request.getType() == null) {
+      throw ServerCommonException.invalidArgument("request.type");
+    }
```

Comment:
tenant id also required

Reply:
Why? Can't I create an event if I'm B2C and have no tenants?